### ISSUE TITLE
General 'web monetization is/is not supported' message needed

### ISSUE THREAD

**Issue by pp-koch** [WICG/webmonetization#246]
So I wrote a really simple example script that plays a video and shows premium content onmonetization. https://quirksmode.org/coil/testsuite/example-script.html

Requires CoilPreview 0.0.70. Only one <link> is supported here, so wait for the video to play, then use the button to move the link tag.

One conclusion is obvious: we MUST have a simple way for the web developer to figure out immediately if web monetization is possible at all. 

If it isn't there should be a message 'Enabled web monetization to view video' or something. That text is not a default placeholder, though: if monetization IS supported but not yet running nothing should happen.

So we need a replacement for

if (!navigator.monetization) {
  // monetization not supported
}

**Comment by sublimator** [comment]
Something else I pondered might warrant `Return of the Monetization` is some kind of field to do provider attestation ... 
but then I wondered if that could just be a field on the monetization event ?

**Comment by sublimator** [comment]
re: CoilPreview
https://chrome.google.com/webstore/detail/coilpreview/iehmfkldnblennopinmmagfidpflefkp

**Comment by marcoscaceres** [comment]
A developer can now check: `if ("MonetizationEvent" in window)` instead of navigator.

**Comment by pp-koch** [comment]
Ugly. But it works. This is something that has to be improved in the next version.

**Comment by marcoscaceres** [comment]
Alternative:
```JS 
var link = document.querySelector("link[rel='monetization']");
if (link.relList.supports("manifest")) {
  // ...supported... 
} 
```

**Comment by sublimator** [comment]
manifest?

**Comment by marcoscaceres** [comment]
Oops, copy/pasta

**Comment by marcoscaceres** [comment]
Fixed

**Comment by sublimator** [comment]
So that would need to be gorilla glued in the coil extension too I guess

**Comment by pp-koch** [comment]
Yes. Right now it doesn't work with the CoilPreview 0.0.72 extension: it always returns false.

**Comment by pp-koch** [comment]
About  (link.relList.supports("monetization")) as a feature detection ...

IF a browser natively supports web monetization
BUT the user is not signed in to any payment provider
will this return true or false?

I assume we can't tell yet, but if we use this line as a general feature detection and it starts to misbehave later we have a problem ...

**Comment by AlexLakatos** [comment]
I think the "browser" related checks (MonetizationEvent, relList) should tell a page that the User Agent (browser or extension) supports monetization, but not if the User is Web Monetized (has a Web Monetization provider / payment pointer and can pay). The triggering of the load event on the link is what informs the page that the User is Web Monetized. And that's because if the User Agent doesn't have a way to pay, it won't process the payment pointer found on the page.

**Comment by MichaelKohler** [comment]
Currently we are using `supports` to detect if the agent is supporting Web Monetization. For the `load` event it currently indicates whether a wallet address (`href`) is valid, and you would get an `error` event otherwise. However this does not indicate that the user has connected a wallet. I'm closing this issue, as there is https://github.com/WICG/webmonetization/issues/502 to discuss the semantics of the load event.
